File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    var answer = 0;
    const N = numbers.length;
    const set = new Set();
    const visited = Array.from({length: N}, ()=> false);
    
    // Permutation
    function dfs(s){
        if(s.length>0)  set.add(Number(s));
        if(s.length===N)    return;

        for(let i=0; i<N; i++){
            if(visited[i])  continue;
            visited[i] = true;
            dfs(s+numbers[i]);
            visited[i] = false;
        }
    }
    dfs("")
    
    const numberArr = [...set];
    
    for(const n of numberArr){
        if(isPrime(n)){
            answer++;
        }
    }
    
    // 소수 찾기
    function isPrime(k){
        if(k<2)   return false;
        for(let i=2; i*i <= k; i++){
            if(k%i===0) return false;
        }
        
        return true;
    }
    
    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

def solution(numbers):
    answer = 0
    
    nums = set()
    
    for i in range(1, len(numbers)+1):
        perm = permutations(list(numbers), i)
        
        for p in perm:
            num = int(''.join(p))
            nums.add(num)
    
    for n in nums:
        if is_prime(n) == True:
            answer += 1
    
    
    return answer


def is_prime(n):
    if n < 2:
        return False
    
    for i in range(2, int(math.sqrt(n))+1):
        if n % i == 0:
            return False
    
    return True
        